### DISCUSSION TITLE
Station List and finding the station closest to coordinates

### DISCUSSION THREAD

**Discussion by jaimefuhr** [weather-gov/api#619] [Q&A] [answered]
I am using the API to get the current temp and conditions by using the browser's location.

I pass the coordinates to get the observationStations.   Then I choose the first station on the list to get obervations/latest.

How is the observationStations list indexed?  Is it in order of closest to the coordinates or is it random?

**Comment by lemonprogis** [comment]
I came here to ask the exact same question. I was wondering if I have to do some type of euclidean distance to see which one is the closest or if it was already ordered closest to furthest.

**Reply by lemonprogis** [reply]
@scadergit, could you provide some input on this? Appreciate it!

**Comment by dwhitemv25** [comment, marked as answer]
Checking for my gridpoint, the list of stations in the [/gridpoints/{wfo}/{x},{y}/stations](https://www.weather.gov/documentation/services-web-api#/default/gridpoint_stations) call are returned in order of increasing distance from the gridpoint.

**Reply by lemonprogis** [reply]
Thank you for confirming that. I actually did the math against each point and it seemed to line up!

**Reply by jaimefuhr** [reply]
So it is confirmed that the first station on the list at index is 0 is the closest and that the list is sorted from nearest to farthest.

I will go ahead and accept it as the answer.

Thanks!

**Comment by scadergit** [comment]
Sorry there was not an earlier answer to this. That is correct. It is in
order of nearest to farthest.
